guides/source/active_record_encryption.md

```diff
@@ -141,199 +252,233 @@ When enabled, all the encryptable attributes will be encrypted according to the
+You can check whether a given attribute is encrypted:
 
-To support these situations, you can declare previous encryption schemes that will be used in two scenarios:
+```ruby
+article.encrypted_attribute?(:title)
+```
 
-* When reading encrypted data, Active Record Encryption will try previous encryption schemes if the current scheme doesn't work.
-* When querying deterministic data, it will add ciphertexts using previous schemes so that queries work seamlessly with data encrypted with different schemes. You must set `config.active_record.encryption.extend_queries = true` to enable this.
+You can read the `cipertext` for an attribute:
 
-You can configure previous encryption schemes:
+```ruby
+article.ciphertext_for(:title)
+```
 
-* Globally
-* On a per-attribute basis
+## Migrating Existing Data
 
-#### Global Previous Encryption Schemes
+### Support for Unencrypted Data
 
-You can add previous encryption schemes by adding them as a list of properties using the `previous` config property in your `application.rb`:
+To ease the transition from unencrypted to encrypted attributes in your Rails application, you can enable support for unencrypted data with:
 
 ```ruby
-config.active_record.encryption.previous = [ { key_provider: MyOldKeyProvider.new } ]
+config.active_record.encryption.support_unencrypted_data = true
 ```
 
-#### Per-attribute Encryption Schemes
+When enabled:
 
-Use `:previous` when declaring the attribute:
+* Reading attributes that are still unencrypted will succeed without raising errors.
+
+* Queries on deterministically encrypted attributes can match both encrypted and cleartext values, if you also enable `extended_queries`:
 
 ```ruby
-class Article
-  encrypts :title, deterministic: true, previous: { deterministic: false }
-end
+config.active_record.encryption.extend_queries = true
 ```
 
-#### Encryption Schemes and Deterministic Attributes
+This setup is intended only for migration periods during which both encrypted and unencrypted data need to coexist in your application. Both options default to `false`, which is the recommended long-term configuration to ensure data is fully encrypted and enforced.
 
-When adding previous encryption schemes:
+### Support for Previous Encryption Schemes
 
-* With **non-deterministic encryption**, new information will always be encrypted with the *newest* (current) encryption scheme.
-* With **deterministic encryption**, new information will always be encrypted with the *oldest* encryption scheme by default.
+Changing encryption properties of attributes can break existing data. For example, imagine you want to make a deterministic attribute non-deterministic. If you simply change the declaration in the model, reading existing ciphertexts will fail because the encryption method is different now.
```

Comment:
(nitpick):

```suggestion
Changing encryption properties of attributes can break existing data. For example, imagine you want to make a deterministic attribute non-deterministic. If you change the declaration in the model, reading existing ciphertexts will fail because the encryption method is different now.
```